ldms/src/contrib/sampler/variorum_sampler/Plugin_variorum_sampler.man

```diff
@@ -0,0 +1,65 @@
+.\" Manpage for Plugin_variorum_sampler
+.\" Contact j_hannebert@coloradocollege.edu, patki1@llnl.gov, shoga1@llnl.gov, brink2@llnl.gov or rountree4@llnl.gov to correct errors or typos.
+.TH man 7 "27 Jun 2022" "v4" "LDMS Plugin variorum_sampler man page"
+
+.SH NAME
+Plugin_variorum_sampler - man page for the LDMS Variorum plugin
+
+.SH SYNOPSIS
+Within ldmsd_controller or a configuration file:
+.br
+config name=variorum_sampler [common attributes]
+
+.SH DESCRIPTION
+With LDMS (Lightweight Distributed Metric Service), plugins for the ldmsd (ldms daemon) are configured via ldmsd_controller
+or a configuration file. The variorum_sampler plugin provides power data using the JSON API in Variorum, a vendor-neutral library that provides access to low-level hardware knobs.
+The sampler, when configured, automatically detects the number of sockets on the host machine and then provides, for each socket, an LDMS record containing power data.
+For each socket, the values provided are: node power consumption in Watts (identical across sockets); socket ID number; CPU power consumption in Watts;
```

Comment:
I think we need more of an explanation about what is going on here, because at first read is sounds a little fishy.

What exactly is a "socket" in your nomenclature? My guess would be a CPU socket.

The first thing that I don't necessarily understand is why things like the entire node's power would be grouped and repeated for each socket. A node is just one node. So shouldn't there just be a single "node_power" field in the schema?

And there is absolutely no reason to assume that the number of cpu sockets matches the number of gpus. So why would we list "gpu_power" per socket?

Is gpu_power the aggregate power used by _all_ gpus or just one gpu's power?

At a very least, I would think we need one list for cpus, and antoher list for gpus.
We might need a list for memory too if there isn't a single memory power number (maybe we have per-dimm or per bank memory power numbers?)

Reply:
Variorum's JSON API has been defined such that it includes all the possible power domains (e.g., CPU, GPU, node, memory) that we may encounter when running variorum on a given HW architecture. Depending on the platform, some of the fields that we have defined in the JSON object will not be populated if the architecture does not provide us with a sensor or register to read the value. As an example, the JSON object when running on lassen will report the node power, the CPU power, and then to answer your question about gpu_power, it is the aggregated power across both GPUs on the socket. 

As of now, Variorum does not yet support two architectures at the same time, so for the lassen case, we cannot report GPU measurements from NVIDIA as well as CPU measurements from IBM. Hope this provides some background on why we have defined the schema in this way, happy to hop on a call with all of us to discuss the details further.